### PR TITLE
DPP-726 Add string interning unit tests

### PR DESCRIPTION
**Pull request by rautenrieth-da**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-integration-api/src/test/suite/scala/platform/store/interning/StringInterningDomainSpec.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        stringToId = entries.map(_.swap).toMap,
+      )
+    }
+  }
+
+  it should "handle a known string " in {
+    val accessor = StaticStringInterningAccessor(List(1 -> ".one", 2 -> ".two"))
+    val domain = StringInterningDomain.prefixing(".", accessor, StringBox.from, StringBox.to)
+
+    domain.tryExternalize(2) shouldBe Some(StringBox("two"))
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/store/interning/StringInterningDomainSpec.scala:42]: can you please extend the tests to the unsafe.*? (in all cases)

File: ledger/participant-integration-api/src/test/suite/scala/platform/store/interning/StringInterningDomainSpec.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  it should "handle an unknown string" in {
+    val accessor = StaticStringInterningAccessor(List(1 -> ".one", 2 -> ".two"))
+    val domain = StringInterningDomain.prefixing(".", accessor, StringBox.from, StringBox.to)
+
+    domain.tryExternalize(3) shouldBe empty
+    domain.tryInternalize(StringBox("three")) shouldBe empty
+  }
+
+  it should "work when two domains share an accessor" in {
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/store/interning/StringInterningDomainSpec.scala:57]: can you please add a test when two domains share one accessor and they have name collision?

File: ledger/participant-integration-api/src/test/suite/scala/platform/store/interning/RawStringInterningSpec.scala [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      Map(1 -> "one", 2 -> "two"),
+      2,
+    )
+    val current = RawStringInterning.from(List(3 -> "three"), previous)
+    current.map shouldBe Map("one" -> 1, "two" -> 2, "three" -> 3)
+    current.idMap shouldBe Map(1 -> "one", 2 -> "two", 3 -> "three")
+    current.lastId shouldBe 3
+  }
+
+  behavior of "RawStringInterning.newEntries"
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/store/interning/RawStringInterningSpec.scala:47]: can you please add one more bigger involved case (like having 3 elements already, adding 5 which from 3 are redundant, and observing new items with correctly assigned id-s)

**Review by nmarton-da — COMMENTED** [review]
looks good! few comments to extend coverage a little bit more

**Review by kamil-da — APPROVED** [review]
LGTM!

**Pull request merged**